### PR TITLE
Batteries included

### PR DESCRIPTION
**Pull request by bclinkinbeard**
This enables use from vanilla projects not created with the generator by taking advantage of the newly built-in components and removing a couple of assumptions about project structure. Using this code you can do this in a fresh, blank project:

`require('idyll')({ inputFile: 'input.idl' }).build()`

This also fixes a "bug" introduced in #65 where it was assumed that source projects would have a dependency on `react-syntax-highlighter`.

### CODE REVIEW

**Comment by rreusser** [comment]
Apologies if it's outside the concern of this PR, but is it possible/reasonable for `inputFile` to be a string or a stream or do you have to back up and assemble the pieces yourself if you want to go to that level?

**Comment by bclinkinbeard** [comment]
@rreusser you can provide a string as `inputString` but we don't currently support streaming.

**Comment by bclinkinbeard** [comment]
Latest commit updates the CLI to match the new (lack of) default options. This means that, if you had `idyll` installed globally, you could simply create an `index.idl` file and run `idyll index.idl --watch` and be off to the races. This is the final piece in making #55 a reality!

**Comment by rreusser** [comment]
Is it possible to modify [this section](https://github.com/idyll-lang/idyll/blob/d2ee8f79e8725954968e5875f3e89be402ca951d/src/index.js#L47-L55) to proceed gracefully if there's no `package.json`? In fact it seems like it'd be better to pass the `package.json` content through options instead of requiring a local file from within this section.

It still fails for me without `package.json`, so this seems like perhaps the last modification necessary to make it truly batteries included.

**Comment by bclinkinbeard** [comment]
@rreusser Good catch. I didn't understand the part about passing it around but I added a check to make `package.json` optional.

File: src/index.js [hunk 1]
```diff
@@ -44,7 +45,7 @@ const idyll = (options = {}, cb) => {
 
   const paths = pathBuilder(opts);
 
-  const inputPackage = require(paths.PACKAGE_FILE);
+  const inputPackage = fs.existsSync(paths.PACKAGE_FILE) ? require(paths.PACKAGE_FILE) : {};
```
Comment (rreusser) [inline src/index.js]: Thanks! 🎉

**Comment by rreusser** [comment]
@bclinkinbeard I was wondering what it would mean to use idyll client-side. Part of that is collecting all the fs operations into one location so the other parts don't depend on a local fs, but this file probably doesn't apply at all if you're on the client.

**Comment by bclinkinbeard** [comment]
Agreed that quarantining all `fs` ops would be a good idea, it made me cringe to add it there. I'm OK with kicking that can down the road though since actually eliminating the `fs` dep is a huge ordeal far off in the future. ¯\_(ツ)_/¯

**Comment by rreusser** [comment]
Yup, no worries here. 😄 It has a hard dependency on fs access anyway, so nothing lost IMO. Wondering if it would someday be possible to write `require('idyll')({ inputString: '...' }).build()` and get the output without fs needing to assume anything about fs access, but yeah, not a goal currently, I think.

**Comment by mathisonian** [comment]
I'm good to merge this as soon as I can figure out how to get the tests to pass on windows

**Comment by mathisonian** [comment]
I'll update the generator to explicitly set the `defaultComponents` path, and we should probably pull the `_index.html` from Idyll now for consistency.

**Pull request merged**